components/raftstore-v2/src/batch/store.rs

```diff
@@ -639,9 +644,22 @@ impl<EK: KvEngine, ER: RaftEngine> StoreSystem<EK, ER> {
                             return;
                         }
                         warn!(logger, "flushing oldest cf of regions {regions:?}");
+                        let flush_count = EK::get_accumulated_flush_count(None).unwrap();
+                        let now = TiInstant::now_coarse();
+                        let total_flush_rate = (flush_count - last_flush.0) as f64
+                            / now.saturating_duration_since(last_flush.1).as_secs_f64();
+                        let dynamic_max_rate = max_rate.clamp(total_flush_rate, f64::MAX);
+                        last_flush = (flush_count, now);
                         // Try to finish flush in 1m.
                         let rate = regions.len() as f64 / MAX_MANUAL_FLUSH_PERIOD.as_secs_f64();
-                        let rate = rate.clamp(MIN_MANUAL_FLUSH_RATE, max_rate);
+                        slog::info!(
+                            logger,
+                            "debug_manual_flush";
+                            "flush_rate" => total_flush_rate,
+                            "dynamic_max_rate" => dynamic_max_rate,
+                            "rate" => rate
+                        );
+                        let rate = rate.clamp(MIN_MANUAL_FLUSH_RATE, dynamic_max_rate);
                         limiter.set_speed_limit(rate);
                         // Return early if there're too many regions.
                         regions.truncate((rate * MAX_MANUAL_FLUSH_PERIOD.as_secs_f64()) as usize);
```

Comment:
There's a problem of truncating regions if some very old cfs may be truncated. 
Is it better to check them all and skip the next tick if conflict?

Reply:
In that case they will be reported again by raft engine. The idea here is to refresh the list often because there's a high chance they are already flushed by write buffer manager.